docs_src/design/adr/0020-UnitOfMeasure.md

```diff
@@ -0,0 +1,93 @@
+
+For example, if the device resource for temperature was specified in a device profile as show below, the unit of measure would still be specified by the units property (specifying the Cel as the unit of measure) and adding the optional units_standard (to specify that the unit of measure is from the Unified Code for Units of Measure).
+
+``` YAML
+-
+  name: "RoomTemperature"
+  isHidden: false
+  description: "Room Temperature x10 °C (Read Only)"
+  attributes:
+    { primaryTable: "INPUT_REGISTERS", startingAddress: 3 }
+  properties:
+    valueType: "Float32"
+    readWrite: "R"
+    scale: "1"
+  units: "Cel"
+  units_standard: "UCUM"
+```
+   
+The unit of measure standard would be specified as a string in the device profile.  Note each device resource could specify a different unit of measure standard.  
+
+### Validation
+
+Initially, EdgeX would not validate the existence of the specification or standard used to specify the unit of measures.  Nor would EdgeX validate that the unit of measure value is a valid unit of measure per that specification.  The unit of measure specification is provided as additional (and optional) information which can be utilized in a way seen fit by the adopter.
+
+EdgeX may, in the future, provide a means to trigger validation of the unit of measure specification and of the unit of measures per that specification.  Doing so would require that EdgeX also have information about where to validate the specification's existence and valid units of measure (presumably a REST URL or other such confirmation site).
+
+In theory, the unit of measure could be validated on the "indbound" collection of the data (likely in a device service), on the "outbound" export of the data from EdgeX (likely in an application service), or in multiple locations if warranted.
+
+## Considersations
+
+It has been suggested that EdgeX borrow from the Open [Geospatial Consortium SensorThings](https://www.ogc.org/standards/sensorthings) standard and include a JSON object to provide more information about the UoM for the field.  An optional  unitOfMeasurement field would replace the optional units_standard in above (allowing for either standard or non-standard definitions but with more information).
+
+A JSON Object containing three key-value pairs. The name property presents the full name of the unitOfMeasurement; the symbol property shows the textual form of the unit symbol; and the definition contains the URI defining the unitOfMeasurement.
+
+From Example 4: A Datastream entity example:
+
+"unitOfMeasurement": {
+    "name": "degree Celsius",
+    "symbol": "°C",
+    "definition": "http://unitsofmeasure.org/ucum.html#para-30"
+}
+
+See https://docs.ogc.org/is/18-088/18-088.html#datastream
+
+## Consequences
+- Any validation could impact performance.
```

Comment:
Unless the validation is performed in Core Metadata as profiles are created/modified.

Reply:
but the validation of the actual values for the unit of measure in a reading would presumably be done by device service or core data or app service and this could get very taxing.  In other words, checking that C or F (for temperatures) is in the list of valid values for that unit of measure.